### DISCUSSION TITLE
Project Settings Extending

### DISCUSSION THREAD

**Discussion by dstoyanoff** [directus/directus#7693] [Feature Requests]
Hey guys, I got a weird idea, which would be very useful. Wanted to hear your thoughts about it.

**Motivation:**
In every project, there are always some basic settings that people would want to have. This is very close to what exists in **directus_settings** today. You basically get instance-level configuration. In order to achieve that today, we need to make a custom settings singleton collection, which means that now we have settings in 2 places - Project settings and custom Settings. 

**Experiment:**
This actually works today. I can add new columns in **directus_settings**, describe them in **directus_fields** and they would appear in the UI, but there are several issues with that:

- you can't really control where the field would appear and perhaps assign it to an existing group
- this is kind of breaking the architecture principals of Directus. The business-related data should not be mixed with the Directus system data, so the solution would need to be a bit different.

**Suggestion:**
Perhaps a possible idea is to have this as part of a metadata file that sits in the repository. This would however be limited to only dev-level customers, so it needs to also be present somehow in the UI.

Any thoughts and feedback would be useful!

**Comment by rijkvanzanten** [comment]
> Perhaps a possible idea is to have this as part of a metadata file that sits in the repository. This would however be limited to only dev-level customers, so it needs to also be present somehow in the UI.

What do you mean by this exactly?

**Reply by dstoyanoff** [reply]
Well, initially I was thinking about including it in the "state file" that we already discussed in the migration steps. Or have a separate file just for the project-level settings. However, this alone will only cover the use-case from developer perspective for clients that work with SCM to manage their Directus. For people using the UI only, there needs to be another mechanism, so that it is controlled from the UI (perhaps internally can still be saved to the same state file).

Hope it is more clear now.

**Reply by rijkvanzanten** [reply]
Ahh I see. I think adding custom fields to directus_settings through the app GUI is perfectly adequate for that purpose. The benefit of that is that it could be included in the state file as well automatically, through the (soon to exist) schema migration flow 🙂

**Reply by dstoyanoff** [reply]
Oh, that's already available if you show the system collections in the settings, my bad 🤦‍♂️

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
